File: daima/19/ModalEditor-Obj/ModalEditor/EditorViewController.m
//
//  EditorViewController.m
//  ModalEditor


#import "EditorViewController.h"

@implementation EditorViewController
@synthesize emailField;

- (id)initWithNibName:(NSString *)nibNameOrNil bundle:(NSBundle *)nibBundleOrNil
{
    self = [super initWithNibName:nibNameOrNil bundle:nibBundleOrNil];
    if (self) {
        // Custom initialization
    }
    return self;
}

- (void)didReceiveMemoryWarning
{
    // Releases the view if it doesn't have a superview.
    [super didReceiveMemoryWarning];
    
    // Release any cached data, images, etc that aren't in use.
}

#pragma mark - View lifecycle

/*
// Implement loadView to create a view hierarchy programmatically, without using a nib.
- (void)loadView
{
}
*/


// Implement viewDidLoad to do additional setup after loading the view, typically from a nib.
// Implement viewDidLoad to do additional setup after loading the view, typically from a nib.
- (void)viewDidLoad
{
    self.emailField.text=((ViewController *)self.presentingViewController).emailLabel.text;
    [super viewDidLoad];
}


- (void)viewDidUnload
{
    [self setEmailField:nil];
    [super viewDidUnload];
    // Release any retained subviews of the main view.
    // e.g. self.myOutlet = nil;
}

- (BOOL)shouldAutorotateToInterfaceOrientation:(UIInterfaceOrientation)interfaceOrientation
{
    // Return YES for supported orientations
    return (interfaceOrientation == UIInterfaceOrientationPortrait);
}

- (IBAction)dismissEditor:(id)sender {
    ((ViewController *)self.presentingViewController).emailLabel.text=self.emailField.text;
    [self dismissViewControllerAnimated:YES completion:nil];
}
@end

File: daima/19/ModalEditor-Swift/ModalEditor/EditorViewController.swift
//
//  EditorViewController.swift


import UIKit

class EditorViewController: UIViewController {
    
    @IBOutlet weak var emailField: UITextField!
    
    @IBAction func updateEditor(_ sender: AnyObject) {
        (presentingViewController as! ViewController).emailLabel.text=emailField.text
        //self.dismissViewControllerAnimated(true, completion: nil)
    }
    
    @IBAction func hideKeyboard(_ sender: AnyObject) {
        emailField.resignFirstResponder()
    }
    
    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.
        preferredContentSize = CGSize(width: 340,height: 160)
    }
    
    override func viewWillAppear(_ animated: Bool) {
        emailField.text=(presentingViewController as! ViewController).emailLabel.text
        super.viewWillAppear(animated)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    


}
